primitives/transaction/src/account/basic_account.rs

```diff
@@ -40,6 +48,14 @@ impl AccountTransactionVerification for BasicAccountVerifier {
     fn verify_outgoing_transaction(transaction: &Transaction) -> Result<(), TransactionError> {
         assert_eq!(transaction.sender_type, AccountType::Basic);
 
+        if !transaction.sender_data.is_empty() {
+            error!(
+                "The following transaction can't have sender data:\n{:?}",
+                transaction
+            );
```

Comment:
Same comment here about the need for an error log. It's not faulty behavior of the node.